packages/block-editor/src/components/block-title/index.js

```diff
@@ -66,6 +66,7 @@ export default function BlockTitle( { clientId } ) {
 	const blockInformation = useBlockDisplayInformation( clientId );
 	if ( ! name || ! blockInformation ) return null;
 	const blockType = getBlockType( name );
+	if ( ! blockType ) return null;
```

Comment:
Reading further it looks like we have a few other fallbacks, so an early return might not make sense here. How about something like:

```jsx
const blockLabel = blockType ? getBlockLabel( blockType, attributes ) : null;
const label = reusableBlockTitle || blockLabel;
// Label will fallback to the title if no label is defined for the current
// label context. If the label is defined we prioritize it over possible
// possible block variation title match.
if ( label && label !== blockType?.title ) {
	return truncate( label, { length: 35 } );
}
return blockInformation.title;
```